test/NuGet.Core.Tests/NuGet.ProjectModel.Test/JsonPackageSpecReaderTests.cs

```diff
@@ -3638,6 +3639,111 @@ public void GetPackageSpec_WithRestoreAuditProperties_ReturnsRestoreAuditPropert
+    ""projectPath"": ""C:\\Users\\me\\source\\code\\project.csproj"",
+    ""projectJsonPath"": ""C:\\Users\\me\\source\\code\\project.json"",
+    ""packagesPath"": ""C:\\Users\\me\\.nuget\\packages"",
+    ""outputPath"": ""C:\\Users\\me\\source\\code\\obj"",
+    ""projectStyle"": ""PackageReference"",
+    ""crossTargeting"": true,
+    ""configFilePaths"": [
+        ""C:\\Users\\me\\source\\code\\NuGet.Config"",
+        ""C:\\Users\\me\\AppData\\Roaming\\NuGet\\NuGet.Config"",
+        ""C:\\Program Files (x86)\\NuGet\\Config\\Microsoft.VisualStudio.FallbackLocation.config"",
+        ""C:\\Program Files (x86)\\NuGet\\Config\\Microsoft.VisualStudio.Offline.config""
+    ],
+    ""fallbackFolders"": [
+        ""C:\\Program Files\\dotnet\\sdk\\NuGetFallbackFolder"",
+        ""C:\\Users\\me\\fallbackFolder""
+
+
+    ]
+  }
+}";
+            var environmentReader = new TestEnvironmentVariableReader(new Dictionary<string, string>()
+                {
+                    { MacroStringsUtility.NUGET_ENABLE_EXPERIMENTAL_MACROS, "true" }
+            });
+
+            PackageSpec actual = PackageSpecTestUtility.GetPackageSpec(json, environmentReader);
+
+            // Assert
+            var metadata = actual.RestoreMetadata;
+
+            Assert.NotNull(metadata);
+            metadata.ProjectUniqueName.Should().Be(@"C:\Users\me\source\code\project.csproj");
+            metadata.ProjectPath.Should().Be(@"C:\Users\me\source\code\project.csproj");
+            metadata.ProjectJsonPath.Should().Be(@"C:\Users\me\source\code\project.json");
+            metadata.PackagesPath.Should().Be(@"C:\Users\me\.nuget\packages");
+            metadata.OutputPath.Should().Be(@"C:\Users\me\source\code\obj");
+
+            metadata.ConfigFilePaths.Should().Contain(@"C:\Users\me\source\code\NuGet.Config");
+            metadata.ConfigFilePaths.Should().Contain(@"C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config");
+            metadata.ConfigFilePaths.Should().Contain(@"C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config");
+            metadata.ConfigFilePaths.Should().Contain(@"C:\Users\me\AppData\Roaming\NuGet\NuGet.Config");
+
+            metadata.FallbackFolders.Should().Contain(@"C:\Program Files\dotnet\sdk\NuGetFallbackFolder");
+            metadata.FallbackFolders.Should().Contain(@"C:\Users\me\fallbackFolder");
+        }
+
+        [Fact]
+        public void GetPackageSpec_RestoreMetadataWithMacros()
+        {
+            // Arrange
+            var json = @"{  
+                            ""restore"": {
+    ""projectUniqueName"": ""C:\\users\\me\\source\\code\\project.csproj"",
+    ""projectName"": ""project"",
+    ""projectPath"": ""C:\\users\\me\\source\\code\\project.csproj"",
+    ""projectJsonPath"": ""C:\\users\\me\\source\\code\\project.json"",
+    ""packagesPath"": ""$(User).nuget\\packages"",
+    ""outputPath"": ""C:\\users\\me\\source\\code\\obj"",
```

Comment:
here, only `packagePath` uses `$(User)`.  Should all of the others as well?

they're not getting validated in the asserts, but given the name of the test method, it's not clear why these other properties are not using the macro, when it feels like they should be.

Reply:
Same as the other one.

Originally, I had all of these as well, but then I realized I needed to change projects refs and all of that as well, and that simply was more than the minimal change dev box needs given that the project is always in the same location. 

msbuild/project system caches depend on that too.